frontend/src/metabase/dashboard/actions/writeback.ts

```diff
@@ -256,16 +290,12 @@ export const executeRowAction = async ({
       parameters,
     });
 
-    if (result["rows-affected"] > 0 || result["rows-updated"]?.[0] > 0) {
-      message = t`Successfully executed the action`;
-    } else if (result["created-row"]) {
-      message = t`Successfully saved`;
-    } else if (result["rows-deleted"]?.[0] > 0) {
-      message = t`Successfully deleted`;
-    } else {
```

Comment:
Are we losing some granularity by not using `rows-affected`, `rows-updated`, `created-row` or `rows-deleted` for custom actions?

Reply:
Can't think of a scenario so far. For explicit actions, we currently show a generic message like "{ACTION_NAME} ran successfully" because it's hard to guess the suitable copy for custom actions for now. For implicit actions, I think we should be okay relying on slug only; if an implicit action fails, we shouldn't get a 2xx response, and the `catch` block of `executeRowAction` should figure out the error message